tests/integration/targets/elb_application_lb_info/tasks/main.yml

```diff
@@ -1,11 +1,250 @@
+        name: '{{ resource_prefix }}_vpc'
+        state: present
+        ipv6_cidr: true
+        tags:
+          Name: elb_application_lb testing
+          ResourcePrefix: "{{ resource_prefix }}"
+      register: vpc
+
+    - name: 'Set fact: VPC ID'
+      set_fact:
+        vpc_id: "{{ vpc.vpc.id }}"
+
+    - name: Create an internet gateway
+      ec2_vpc_igw:
+        vpc_id: '{{ vpc_id }}'
+        state: present
+        tags:
+          Name: '{{ resource_prefix }}'
+      register: igw
+
+    - name: Create private subnets
+      ec2_vpc_subnet:
+        cidr: '{{ item.cidr }}'
+        az: '{{ aws_region }}{{ item.az }}'
+        vpc_id: '{{ vpc_id }}'
+        state: present
+        tags:
+          Public: 'False'
+          Name: 'private-{{ item.az }}'
+      with_items:
+        - cidr: "{{ private_subnet_cidr_1 }}"
+          az: a
+        - cidr: "{{ private_subnet_cidr_2 }}"
+          az: b
+      register: private_subnets
+
+    - name: Create public subnets with ipv6
+      ec2_vpc_subnet:
+        cidr: '{{ item.cidr }}'
+        az: '{{ aws_region }}{{ item.az }}'
+        vpc_id: '{{ vpc_id }}'
+        state: present
+        ipv6_cidr: '{{ item.vpc_ipv6_cidr }}'
+        tags:
+          Public: 'True'
+          Name: 'public-{{ item.az }}'
+      with_items:
+        - cidr: "{{ public_subnet_cidr_1 }}"
+          az: a
+          vpc_ipv6_cidr: "{{ vpc.vpc.ipv6_cidr_block_association_set[0].ipv6_cidr_block | replace('0::/56','0::/64') }}"
+        - cidr: "{{ public_subnet_cidr_2 }}"
+          az: b
+          vpc_ipv6_cidr: "{{ vpc.vpc.ipv6_cidr_block_association_set[0].ipv6_cidr_block | replace('0::/56','1::/64') }}"
+      register: public_subnets
+
+    - name: Create list of subnet ids
+      set_fact:
+        public_subnets: "{{ public_subnets.results | map(attribute='subnet') | map(attribute='id') }}"
+        private_subnets: "{{ private_subnets.results | map(attribute='subnet') | map(attribute='id') }}"
+
+    - name: Create a route table
+      ec2_vpc_route_table:
+        vpc_id: '{{ vpc_id }}'
+        tags:
+          Name: igw-route
+          Created: '{{ resource_prefix }}'
+        subnets: '{{ public_subnets + private_subnets }}'
+        routes:
+          - dest: 0.0.0.0/0
+            gateway_id: '{{ igw.gateway_id }}'
+      register: route_table
+
+    - name: Create a security group for Ansible ELB integration tests
+      ec2_group:
+        name: '{{ resource_prefix }}'
+        description: security group for Ansible ELB integration tests
+        state: present
+        vpc_id: '{{ vpc_id }}'
+        rules:
+          - proto: tcp
+            from_port: 1
+            to_port: 65535
+            cidr_ip: 0.0.0.0/0
+      register: sec_group
+
+    - name: Create a target group for testing
+      elb_target_group:
+        name: '{{ tg_name }}'
+        protocol: http
+        port: 80
+        vpc_id: '{{ vpc_id }}'
+        state: present
+      register: tg
+
```

Comment:
Can you please organise the tests as tremble suggested in this comment https://github.com/ansible-collections/amazon.aws/pull/509#issuecomment-932056400?

Reply:
yes - wasnt sure if it needed that depth on the info module but I will update

Reply:
I mean for the elb_application_lb module, not for the info module (Sorry, pinned the comment on the wrong file).

Reply:
So, I actually missed the tests for elb_application_lb.